src/errors/BadRequestError.ts

```diff
@@ -0,0 +1,30 @@
+import { BaseIsomerError } from "./BaseError"
+import {
+  IdentifiableError,
+  PossibleComponentTypes,
+  PossibleFileCodes,
+} from "./IsomerError"
+
+export default class BadRequestError
+  extends BaseIsomerError
+  implements IdentifiableError {
+  componentCode: PossibleComponentTypes
+
+  fileCode: PossibleFileCodes
+
+  constructor(
+    // NOTE: This is technically optional but we want to
+    // preserve argument ordering for back-compat
+    // with our JS files.
```

Comment:
Hmm I dont understand this comment, what does the argument being optional relate to argument ordering here ya?

Reply:
oops, this comment was written when i set `message` to a default arg. will remove